tests/init_replication.rs

```diff
@@ -537,6 +537,48 @@ fn insertion_after_removal() {
     assert!(client_entity.contains::<TableComponent>());
 }
 
+#[test]
+fn removal_with_despawn() {
+    let mut server_app = App::new();
+    let mut client_app = App::new();
+    for app in [&mut server_app, &mut client_app] {
+        app.add_plugins((
+            MinimalPlugins,
+            ReplicationPlugins.set(ServerPlugin {
+                tick_policy: TickPolicy::EveryFrame,
+                ..Default::default()
+            }),
+        ))
+        .replicate::<TableComponent>()
+        .replicate::<SparseSetComponent>()
+        .replicate::<NotReplicatedComponent>()
+        .replicate_mapped::<MappedComponent>();
+    }
+
+    connect::single_client(&mut server_app, &mut client_app);
+
+    // Remove component first.
+    let server_entity = server_app
+        .world
+        .spawn((Replication, TableComponent))
+        .remove::<TableComponent>()
+        .id();
+    let client_entity = client_app.world.spawn(Replication).id();
```

Comment:
> spawn and replicate TableComponent, then after replicated do a removal + despawn in the same tick

I spawn an entity on client, map it into server one and then check if it was despawned. So this part should be checked: we don't panic when there is a component removal that doesn't exist for existing entity.

I also checked warning, nothing like this.

Reply:
That's what I do, except instead of actually spawning and replicating, I mocking it by spawning an entity on client and server and just map them.

Reply:
Do you think it's confusing and it's better to do actual thing?

Reply:
I went ahead and refactored tests.
But instead of adding a new test, I changed current tests. Doing actual spawning, replicating and only then removal / insertion as you suggested is more clear, I don't think that we need a duplicated t test for doing the same in a single frame.